src/lib/storage/value_segment.cpp

```diff
@@ -1,73 +1,94 @@
 #include "value_segment.hpp"
+#include <optional>
 
 #include "type_cast.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
 template <typename T>
-ValueSegment<T>::ValueSegment(bool nullable) {
-  // Implementation goes here
-}
+ValueSegment<T>::ValueSegment(bool nullable) : _is_nullable(nullable) {}
 
 template <typename T>
 AllTypeVariant ValueSegment<T>::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (is_null(chunk_offset)) {
+    return NULL_VALUE;
+  }
+  return get(chunk_offset);
 }
 
 template <typename T>
 bool ValueSegment<T>::is_null(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return is_nullable() && null_values()[chunk_offset];
 }
 
 template <typename T>
 T ValueSegment<T>::get(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(!is_null(chunk_offset), "Tried to .get a NULL value from a ValueSegment.");
+  return values()[chunk_offset];
 }
 
 template <typename T>
 std::optional<T> ValueSegment<T>::get_typed_value(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (is_null(chunk_offset)) {
+    return std::nullopt;
+  }
+  return get(chunk_offset);
 }
 
 template <typename T>
 void ValueSegment<T>::append(const AllTypeVariant& value) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (variant_is_null(value)) {
+    Assert(is_nullable(), "Tried to append NULL value into non-nullable ValueSegment.");
+    _values.emplace_back();
+    _null_values.emplace_back(true);
+  } else {
+    try {
+      _values.emplace_back(type_cast<T>(value));
```

Comment:
I think after the update of the upstream, you can now get rid of one of the nested try-catch blocks